microraiden/microraiden/test/test_http_client.py

```diff
@@ -536,3 +537,71 @@ def test_requests(
         response = microraiden.requests.get(url)
 
     assert response.text == 'success'
+
+
+def test_error_handling(
+        default_http_client: DefaultHTTPClient,
+        api_endpoint_address: str,
+        token_address: str,
+        channel_manager_address: str,
+        receiver_address: str
+):
+    nonexisting_channel_mock = mock.patch.object(
+        default_http_client,
+        'on_nonexisting_channel',
+        wraps=default_http_client.on_nonexisting_channel
+    ).start()
+    insufficient_confirmations_mock = mock.patch.object(
+        default_http_client,
+        'on_insufficient_confirmations',
+        wraps=default_http_client.on_insufficient_confirmations
+    ).start()
+    insufficient_funds_mock = mock.patch.object(
+        default_http_client,
+        'on_insufficient_funds',
+        wraps=default_http_client.on_insufficient_funds
+    ).start()
+    invalid_contract_address_mock = mock.patch.object(
+        default_http_client,
+        'on_invalid_contract_address',
+        wraps=default_http_client.on_invalid_contract_address
+    ).start()
+
+    with requests_mock.mock() as server_mock:
+        headers = {
+            HTTPHeaders.TOKEN_ADDRESS: token_address,
+            HTTPHeaders.CONTRACT_ADDRESS: channel_manager_address,
+            HTTPHeaders.RECEIVER_ADDRESS: receiver_address,
+            HTTPHeaders.PRICE: '3'
+        }
+        headers = [headers.copy() for i in range(5)]
```

Comment:
or
```
headers = [headers.copy() for _ in range(5)]
headers[1][HTTPHeaders.NONEXISTING_CHANNEL] = '1'
headers[2][HTTPHeaders.INSUF_CONFS] = '1'
headers[3][HTTPHeaders.INSUF_FUNDS] = '1'
headers[4][HTTPHeaders.CONTRACT_ADDRESS] = '0x' + '12' * 20
server_mock.get(url, [ 
  {'status_code': 402, 'headers': header}
  for header in headers
])
```

Reply:
Alright, will switch to that in both PRs.